src/sc2api/sc2_coordinator.cc

```diff
@@ -197,7 +197,9 @@ CoordinatorImp::CoordinatorImp() :
 
 CoordinatorImp::~CoordinatorImp() {
     for (auto& p : process_settings_.process_info) {
-        TerminateProcess(p.process_id);
+        if (p.process_id > 0) {
+            TerminateProcess(p.process_id);
```

Comment:
True. I am trying to think of a case where you would want to use `TerminateProcess(0)` and I can't find any. So I would add this `if...` [here](https://github.com/Blizzard/s2client-api/blob/d9ba0a33d6ce9d233c2a4ee988360c188fbe9dbf/src/sc2utils/sc2_manage_process.cc#L452) instead and return `false`.

But @N00byEdge reacted with a thumps down. So maybe there is a reason to keep it there?

Reply:
If you really want to prevent such behaviour, it should be placed inside `TerminateProcess`.
To me `0` looks like `not properly initialised object`. However, there could be some logic around it or simply a error in the code (i.e. the id is not set when needed) but we don't see it because of the current behaviour of `TerminateProcess`.

Is it covered by unit tests somehow?

@KevinCalderone could you please shed the light on this implementation?